src/com/itacademy/eshop/shop/ShoppingCart.java

```diff
@@ -28,6 +30,12 @@ public Integer getTotalPrice() {
         }
         return totalPrice;
     }
+    // this would make more sense to just return the avg price
+    // but is a setter for the sake of task no. 1 in encapsulation
+    public void setAvgPriceOfItems(){
+        avgPriceOfItemsInCart = getTotalPrice()/products.size();
```

Comment:
Division by `products.size()` might cause `ArithmeticException` because in line 13 products is set as empty list which size is 0 and division from zero is not allowed. You probably need to add additional check to make sure that size not equals zero